igor-web/src/main/java/com/netflix/spinnaker/igor/scm/DeliveryConfigManifestsController.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.scm;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.netflix.spinnaker.igor.config.DeliveryConfigProperties;
+import com.netflix.spinnaker.igor.scm.bitbucket.client.BitBucketMaster;
+import com.netflix.spinnaker.igor.scm.github.client.GitHubMaster;
+import com.netflix.spinnaker.igor.scm.gitlab.client.GitLabMaster;
+import com.netflix.spinnaker.igor.scm.stash.client.StashMaster;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/** Exposes APIs to retrieve Managed Delivery declarative manifests from source control repos. */
+@RestController
+@EnableConfigurationProperties(DeliveryConfigProperties.class)
+@RequestMapping("/delivery-config")
+public class DeliveryConfigManifestsController {
+  private static final Logger log = LoggerFactory.getLogger(AbstractCommitController.class);
+
+  private final DeliveryConfigProperties configProperties;
+  private final ObjectMapper jsonMapper;
+  private final ObjectMapper yamlMapper;
+  private final Optional<StashMaster> stashMaster;
+  private final Optional<GitHubMaster> gitHubMaster;
+  private final Optional<GitLabMaster> gitLabMaster;
+  private final Optional<BitBucketMaster> bitBucketMaster;
+
+  public DeliveryConfigManifestsController(
```

Comment:
Comment on the controller: maybe it's better to move all the logic to a separate class, like DeliveryConfigManifestsService? That will keep the controller clean and will follow the same conventions as we have in other services.

Reply:
I didn’t think there was a logic here to warrant a new service object, plus there’s already some processing in the new `ScmMaster` methods, but if it’s a pattern I’d definitely make that change.

Reply:
You can see gate for example. Its up to you I guess :)

Reply:
I do think it's up to you Luis. I generally like to pull all logic to a separate class and keep the controllers thin. That way it's easier to write tests (and you can avoid testing the controllers unless you'd like to specifically test serialization behavior, etc). Specially, there's a lot of logic in `getManifest(` that would probably belong in a service class.

Reply:
Fair points. I’ll add the service class.

Reply:
Done. Hope that looks better.